src/Draggable/tests/Draggable.test.js

```diff
@@ -31,18 +84,579 @@ describe('Draggable', () => {
+      const newInstance = new Draggable();
+      function stubHandler() { /* do nothing */ }
+
+      expect('my:event' in newInstance.callbacks)
+        .toBe(false);
+
+      newInstance.on('my:event', stubHandler);
+
+      expect('my:event' in newInstance.callbacks)
+        .toBe(true);
+
+      expect(newInstance.callbacks['my:event'])
+        .toMatchObject([stubHandler]);
+    });
+
+    test('should return draggable instance', () => {
+      const newInstance = new Draggable();
+      function stubHandler() { /* do nothing */ }
+
+      expect('my:event' in newInstance.callbacks)
+        .toBe(false);
+
+      const returnValue = newInstance.on('my:event', stubHandler);
+
+      expect(returnValue)
+        .toBe(newInstance);
+    });
+  });
+
+  describe('#off', () => {
+    test('should return null if event was not bound', () => {
+      const newInstance = new Draggable();
+      function stubHandler() { /* do nothing */ }
+
+      expect('my:event' in newInstance.callbacks)
+        .toBe(false);
+
+      const returnValue = newInstance.off('my:event', stubHandler);
+
+      expect(returnValue).toBe(null);
+    });
+
+    test('should remove event handler from the list of callbacks', () => {
+      const newInstance = new Draggable();
+      function stubHandler() { /* do nothing */ }
+
+      newInstance.on('my:event', stubHandler);
+
+      expect('my:event' in newInstance.callbacks)
+        .toBe(true);
+
+      newInstance.off('my:event', stubHandler);
+
+      expect('my:event' in newInstance.callbacks)
+        .toBe(true);
+
+      expect(newInstance.callbacks['my:event'])
+        .toMatchObject([]);
+    });
+
+    test('should return draggable instance', () => {
+      const newInstance = new Draggable();
+      function stubHandler() { /* do nothing */ }
+
+      newInstance.on('my:event', stubHandler);
+
+      expect('my:event' in newInstance.callbacks)
+        .toBe(true);
+
+      const returnValue = newInstance.off('my:event', stubHandler);
+
+      expect(returnValue)
+        .toBe(newInstance);
+    });
+  });
+
+  describe('#trigger', () => {
+    test('should invoke bound event', () => {
+      const handler = jest.fn();
+      const expectedEvent = new Event('my:event');
+
+      draggable.on('my:event', handler);
+
+      draggable.trigger(expectedEvent);
+
+      expect(handler.mock.calls.length)
+        .toBe(1);
+
+      expect(handler.mock.calls[0].length)
+        .toBe(1);
+
+      expect(handler.mock.calls[0][0])
+        .toBe(expectedEvent);
+    });
+  });
+
   test('triggers `drag:start` drag event on mousedown', () => {
     const draggableElement = sandbox.querySelector('li');
     document.elementFromPoint = () => draggableElement;
 
     const callback = jest.fn();
     draggable.on('drag:start', callback);
-    triggerEvent(draggableElement, 'mousedown', {button: 0});
+
+    const event = new MouseEvent('mousedown');
```

Comment:
OK, I believe you can chalk that up to user error. I couldn't recreate the issue I was seeing. It may be that I omitted the `{button: 0}` context for the event's constructor? Not sure. May honestly have been a reaction due to personal preference for directly invoking the event using the event objects' constructors. Is the choice to use an older style of event object initialization intentional? Just curious about the design of `triggerEvent`.

Anyway, I'm open to your suggestion about changing out the raw DOM events with sensor events. Should that go in a new PR, or is this one OK?